### PR TITLE
Initial commit of dpct migration samples.

### PR DESCRIPTION
**Pull request by xmnboy**
Signed-off-by: xmnboy <xmnboy@yahoo.com>

# Description

Three samples being added to demonstrate the use of the dpct migration tool. These samples have been approved and developed in conjunction with @tomlenth, Yury P. and @srdontha.

These three samples are going into a new folder named `Migration` inside the existing `Tools` folder. 

There is no CI section, yet, in the `sample.json` because the CI system will need to be modified in order to perform automated testing on these samples. 

Fixes # (issue) 

## Type of change

- [X] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [X] Visual Studio

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [X] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [X] Implement coding guidelines and ensure code quality.
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [X] Adhere to readme template 
- [X] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [X] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: Tools/Migration/folder-options-dpct/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Copyright Intel Corporation
```
Comment (JoeOster) [inline Tools/Migration/folder-options-dpct/LICENSE.txt:1]: Is there a reason that you are not using the standard oneAPI License file? If not, you should be using this one: https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: Tools/Migration/vector-add-dpct/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Copyright Intel Corporation
```
Comment (JoeOster) [inline Tools/Migration/vector-add-dpct/LICENSE.txt:1]: Is there a reason that you are not using the standard oneAPI License file? If not, you should be using this one: https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: Tools/Migration/rodinia-nw-dpct/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,87 @@
+Copyright Intel Corporation
```
Comment (JoeOster) [inline Tools/Migration/rodinia-nw-dpct/LICENSE.txt:1]: Is there a reason that you are not using the standard oneAPI License file for the Intel aspect? If not, you should be using this one: https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

To keep with consistent formatting match the case, please

**Review by JoeOster — APPROVED** [review]

File: Tools/Migration/folder-options-dpct/LICENSE.txt (hunk 1 above)
Comment (xmnboy) [reply Tools/Migration/folder-options-dpct/LICENSE.txt:1]: It's the MIT license file. It's just been word-wrapped so it's easier to read and we've added the SPDX line and the link referencing the official MIT license file.

File: Tools/Migration/vector-add-dpct/LICENSE.txt (hunk 2 above)
Comment (xmnboy) [reply Tools/Migration/vector-add-dpct/LICENSE.txt:1]: It's the MIT license file. It's just been word-wrapped so it's easier to read and we've added the SPDX line and the link referencing the official MIT license file.

File: Tools/Migration/rodinia-nw-dpct/LICENSE.txt (hunk 3 above)
Comment (xmnboy) [reply Tools/Migration/rodinia-nw-dpct/LICENSE.txt:1]: It's the MIT license file. It's just been word-wrapped so it's easier to read and we've added the SPDX line and the link referencing the official MIT license file.

Also, this sample is derived from an existing open-source project, thus the reason there is the original license file from that original project, as well as the MIT license file that the samples are using.

**Pull request merged**